File: yukicoder/0090.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
  int N, M;

  scanf("%d %d", &N, &M);

  vector<vector<int> > score = vector<vector<int> >(9, vector<int>(9, 0));

  for (int i = 0; i < M; i++) {
    int a, b, c;
    scanf("%d %d %d", &a, &b, &c);
    score[a][b] = c;
  }
  long long int max_score = 0;
  int p[10];

  for (int i = 0; i < N; i++) {
    p[i] = i;
  }

  do {
    int s = 0;

    for (int i = 0; i < N - 1; i++)
      for (int j = i + 1; j < N; j++) s += score[p[i]][p[j]];
    if (s > max_score) max_score = s;
  } while (next_permutation(p, p + N));
  printf("%d\n", max_score);
  return 0;
}

File: yukicoder/0090.py
import itertools

N, M = map(int, input().split())
score = [[0 for j in range(N)] for i in range(N)]
for i in range(M):
    a, b, s = map(int, input().split())
    score[a][b] = s
# max_score = 0
# for p in itertools.permutations(range(N)):
#     s = 0
#     for i in range(N - 1):
#         for j in range(i + 1, N):
#             s += score[p[i]][p[j]]
#     max_score = max(max_score, s)
# print(max_score)

greed = []
for i in range(N):
    max_s = 0
    for j in range(len(greed)+2):
        s = score[]
